src/Covenant/Test.hs

```diff
@@ -835,6 +846,64 @@ shrinkDataDecls decls = liftShrink shrinkDataDecl decls <|> (shrinkDataDecl <$>
 genDataList :: forall (a :: Type). DataGenM a -> Gen [a]
 genDataList = runDataGenM . GT.listOf
 
+-- ASG Stuff
+
+-- | A concrete monadic stack, providing the minimum amount of functionality
+-- needed to build an ASG using the combinators given in this module.
```

Comment:
Slightly misleading docs here, as 'this module' actually doesn't contain its primary source of functionality.